test/unit/bookmark_test.rb

```diff
@@ -20,6 +20,15 @@ class BookmarkTest < ActiveSupport::TestCase
     end
   end
 
+  test "my bookmarks should be able to create two bookmarks with same name under different controllers" do
+    enable_login do
+      assert_difference 'Bookmark.count',2 do
+        Bookmark.create({:name => "private", :query => "bar", :public => false, :controller => "hosts"})
```

Comment:
Use `create!` to pick up creation errors, or even just create one (via FG) and then check the second is valid without saving it, to reduce DB load.